lib/server-args.js

```diff
@@ -0,0 +1,348 @@
+import fs from 'fs';
+import _ from 'lodash';
+import { DEFAULT_BASE_PATH } from 'appium-base-driver';
+
+// serverArgs will be added to the `server` (default) subcommand
+function parseSecurityFeatures (features) {
+  const splitter = (splitOn, str) => `${str}`.split(splitOn).map((s) => s.trim()).filter(Boolean);
+  let parsedFeatures;
+  try {
+    parsedFeatures = splitter(',', features);
+  } catch (err) {
+    throw new Error('Could not parse value of --allow/deny-insecure. Should be ' +
+                    'a list of strings separated by commas, or a path to a file ' +
+                    'listing one feature name per line.');
+  }
+
+  if (parsedFeatures.length === 1 && fs.existsSync(parsedFeatures[0])) {
+    // we might have a file which is a list of features
+    try {
+      const fileFeatures = fs.readFileSync(parsedFeatures[0], 'utf8');
+      parsedFeatures = splitter('\n', fileFeatures);
+    } catch (err) {
+      throw new Error(`Attempted to read --allow/deny-insecure feature names ` +
+                      `from file ${parsedFeatures[0]} but got error: ${err.message}`);
+    }
+  }
+
+  return parsedFeatures;
+}
+
+function parseDefaultCaps (caps) {
+  try {
+    // use synchronous file access, as `argparse` provides no way of either
+    // awaiting or using callbacks. This step happens in startup, in what is
+    // effectively command-line code, so nothing is blocked in terms of
+    // sessions, so holding up the event loop does not incur the usual
+    // drawbacks.
+    if (fs.statSync(caps).isFile()) {
+      caps = fs.readFileSync(caps, 'utf8');
+    }
+  } catch (err) {
+    // not a file, or not readable
+  }
+  caps = JSON.parse(caps);
+  if (!_.isPlainObject(caps)) {
+    throw 'Invalid format for default capabilities';
+  }
+  return caps;
+}
+
+export default [
+  [['--shell'], {
```

Comment:
so how about moving these into a separate JSON config file? such json files could be loaded with simple `import` operator like ES modules.

Reply:
a couple reasons i don't like this:

1) json is like js just with less flexibility and more quotation marks
2) some of these cannot be represented easily in JSON, for example:

```js
  [['-pa', '--base-path'], {
    required: false,
    defaultValue: DEFAULT_BASE_PATH,
    dest: 'basePath',
    example: '/path/prefix',
    help: 'Base path to use as the prefix for all webdriver routes running' +
          `on this server (default: ${DEFAULT_BASE_PATH})`
  }],
```

or

```js
  [['-dc', '--default-capabilities'], {
    dest: 'defaultCapabilities',
    defaultValue: {},
    type: parseDefaultCaps,
    required: false,
    example: '[ \'{"app": "myapp.app", "deviceName": "iPhone Simulator"}\' ' +
             '| /path/to/caps.json ]',
    help: 'Set the default desired capabilities, which will be set on each ' +
          'session unless overridden by received capabilities.'
  }],
```

(Note the use of variables here corresponding to strings and functions)

3) this file is already loaded with a simple `import` operator...

but maybe I'm missing something about your suggestion?

Reply:
no, this makes sense. Let's keep it as is, but isolate the args list itself from the rest of the code